docs/cudf/source/conf.py

```diff
@@ -557,6 +560,67 @@ def on_missing_reference(app, env, node, contnode):
 ]
 
 
+# Needed for the [source] button on the API docs to link to the github code
+# based on pandas doc/source/conf.py
+def linkcode_resolve(domain, info) -> str | None:
+    """
+    Determine the URL corresponding to Python object
+    """
+    if domain != "py":
+        return None
+
+    modname = info["module"]
+    fullname = info["fullname"]
+
+    submod = sys.modules.get(modname)
+    if submod is None:
+        return None
+
+    obj = submod
+    for part in fullname.split("."):
+        try:
+            with warnings.catch_warnings():
+                # Accessing deprecated objects will generate noisy warnings
+                warnings.simplefilter("ignore", FutureWarning)
+                obj = getattr(obj, part)
+        except AttributeError:
+            return None
+
+    try:
+        fn = inspect.getsourcefile(inspect.unwrap(obj))
+    except TypeError:
+        try:  # property
+            fn = inspect.getsourcefile(inspect.unwrap(obj.fget))
+        except (AttributeError, TypeError):
+            fn = None
+    if not fn:
+        return None
+
+    try:
+        source, lineno = inspect.getsourcelines(obj)
+    except TypeError:
+        try:  # property
+            source, lineno = inspect.getsourcelines(obj.fget)
+        except (AttributeError, TypeError):
+            lineno = None
+    except OSError:
+        lineno = None
+
+    if lineno:
+        linespec = f"#L{lineno}-L{lineno + len(source) - 1}"
+    else:
+        linespec = ""
+
+    fn = os.path.relpath(fn, start=os.path.dirname(cudf.__file__))
+    if "+" in cudf.__version__:
+        return f"https://github.com/rapidsai/cudf/blob/main/cudf/{fn}{linespec}"
+    else:
+        return (
+            f"https://github.com/rapidsai/cudf/blob/"
+            f"branch-{version}/python/cudf/cudf/{fn}{linespec}"
+        )
```

Comment:
This check relies on a specific versioning scheme for development versions that pandas uses but cudf does not (see `https://github.com/python-versioneer/python-versioneer/blob/master/details.md#how-do-i-select-a-version-style`). The equivalent in cudf would be to check whether we're building an alpha, but we don't really need that since we presumably always just want the latest branch, so we can just use the else branch unconditionally
```suggestion
    return (
        f"https://github.com/rapidsai/cudf/blob/"
        f"branch-{version}/python/cudf/cudf/{fn}{linespec}"
    )
```